.github/workflows/test-android.yml

```diff
@@ -0,0 +1,130 @@
+name: End To End Tests
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'CODE_OF_CONDUCT.md'
+      - 'CONTRIBUTING.md'
+      - 'LICENSE'
+      - 'SECURITY.md'
+      - 'docs/**'
+      - '.github/**'
+      - '!.github/workflows/test.yml'
+  push:
+    branches:
+      - develop
+      - main
+      - release/*
+    tags: '[0-9]+.[0-9]+.[0-9]+'
+    paths-ignore:
+      - 'README.md'
+      - 'CODE_OF_CONDUCT.md'
+      - 'CONTRIBUTING.md'
+      - 'LICENSE'
+      - 'SECURITY.md'
+      - 'docs/**'
+      - '.github/**'
+      - '!.github/workflows/test.yml'
+  release:
+    types:
+      - created
+
+jobs:
+  end-to-end-tests:
+    name: Android End To End Tests
+    runs-on: [self-hosted, ase-large-android-sdk-34]
+    timeout-minutes: 60
+
+    steps:
+      - name: Checkout Main Artemis Repo (this repo)
+        uses: actions/checkout@v4
+        with:
+          path: main-repo
+
+      - name: Checkout Artemis Android Repo
+        uses: actions/checkout@v4
+        with:
+          repository: ls1intum/artemis-android
+          path: android-repo
+
+        # sanity cleanup
+      - name: Stop mysql server
+        run: docker stop artemis-mysql
+        continue-on-error: true
+
+      - name: Remove mysql server
+        run: docker rm artemis-mysql
+        continue-on-error: true
+
+      - name: Launch mysql docker container
+        run: docker compose -f main-repo/docker/mysql.yml up -d mysql
+
+      - name: JDK setup
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+
+        # Run artemis server in a detached mode, and store the pid in variable steps.run-artemis.outputs.pid
+      - name: Run Artemis Server
+        id: run-artemis
+        run: ./main-repo/gradlew bootRun --args='--spring.profiles.active=dev,artemis,scheduling --artemis.user-management.use-external=false --artemis.user-management.internal-admin.username=artemis_admin --artemis.user-management.internal-admin.password=artemis_admin --artemis.user-management.registration.enabled=true --artemis.user-management.registration.allowed-email-pattern=.*' & echo "pid=$!" >> "$GITHUB_OUTPUT"
```

Comment:
Optional 2: Also afaik this would build Artemis first. Maybe we could consider depending on the build job before starting it (e.g. via a reusable workflow).
Though this would likely be a little too much for now and should ideally be part of a bigger restructuring (possibly by me) in the future, though I just wanted to mention it ^^

Reply:
For now, I would suggest leaving it as it is.